newdle/client/src/components/summary/SummaryPage.js

```diff
@@ -36,14 +54,52 @@ export default function SummaryPage() {
     }
   };
 
-  const sendResultEmails = () => _sendResultEmails(newdle.code);
+  const handleMailModalClose = useCallback(() => {
+    setMailModalOpen(false);
+  }, [setMailModalOpen]);
+
+  const handleMailModalConfirm = useCallback(() => {
+    setMailModalOpen(false);
+    _sendResultEmails(newdle.code);
+  }, [setMailModalOpen, newdle, _sendResultEmails]);
 
   if (!newdle) {
     return <Loader active />;
   }
 
   const mailSent = sendMailResponse !== null;
 
+  const recipientList = (list, color, icon, limit, limitSetter) => {
+    var sliced_list = list.slice(0, limit);
+    return (
+      <List>
+        {sliced_list.map((p, index) => {
+          return (
+            <List.Item key={p.id}>
+              <List.Icon color={color} name={icon} />
+              <List.Content>{p.name}</List.Content>
+            </List.Item>
+          );
+        })}
+        {limit < list.length && (
+          <List.Item>
+            <List.Icon color="black" name="ellipsis horizontal" />
+            <List.Content>
+              <a
+                href="#!"
+                onClick={() => {
+                  limitSetter(list.length);
```

Comment:
Since you switch from "limited" to "everything" you can simplify this by simply storing a bool whether the list is expanded or not